src/api.cr

```diff
@@ -22,6 +23,19 @@ module OpenTelemetry
           ],
         )
       end
+
+      def resource_from_env
+        return nil if ENV["HONEYCOMB_DATASET"]?.nil?
+
+        Proto::Resource::V1::Resource.new(
+          attributes: [
+            Proto::Common::V1::KeyValue.new(
+              key: "service.name",
+              value: Proto::Common::V1::AnyValue.new(ENV["HONEYCOMB_DATASET"])
```

Comment:
The OTel spec [prescribes `OTEL_SERVICE_NAME`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/sdk-environment-variables.md#general-sdk-configuration), and this patch avoids fetching from the ENV multiple times (it's not _slow_, but still we can almost certainly count on it not changing during runtime), and maybe we could even store this in a constant for sub-nanosecond access times. 😂 Not necessary here, to be clear, just thinking out loud.

The most important thing is the env-var name.

```suggestion
        unless service_name = ENV["OTEL_SERVICE_NAME"]?
          return
        end

        Proto::Resource::V1::Resource.new(
          attributes: [
            Proto::Common::V1::KeyValue.new(
              key: "service.name",
              value: Proto::Common::V1::AnyValue.new(service_name)
```

Reply:
`OTEL_SERVICE_NAME` is better in a purely-OpenTelemetry SDK than `HONEYCOMB_DATASET` and I say that as a Honeycomber.

🤔 The reading of resource attributes from the environment—`OTEL_RESOURCE_ATTRIBUTES` and `OTEL_SERVICE_NAME` per [the General SDK Configuration spec](https://opentelemetry.io/docs/reference/specification/sdk-environment-variables/#general-sdk-configuration)—could be done once [at configuration time](https://github.com/jgaskins/opentelemetry/blob/ca32f4a72f0e1fc67d6f2e3a5f4ee90176b08e0e/src/opentelemetry.cr#L111-L113), creating a Resource, setting attributes on it appropriately, and storing it as a property of the `CONFIG` object.

Reply:
Thank you both for the feedback here! I wasn't aware of the OTel spec definition of those ENV vars.

I just updated the PR to use those at configuration time for performance and to make the shard more compliant with the SDK configuration spec.